### DISCUSSION TITLE
[ZK Stack] - How could one set a custom genesis?

### DISCUSSION THREAD

**Discussion by ETeissonniere** [zkSync-Community-Hub/zksync-developers#263] [ZK Stack] [answered]
### Module Affected

- [X] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [X] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.74.0-nightly

### Operating System

- [ ] Ubuntu
- [X] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

I would expect it is possible to set a custom genesis configuration when creating a new hyperchain (ie. what contracts are predeployed, what balances are already set, etc...).  However, I am not immediately clear how to do so, or even if it is supported.

### Expected Behavior

A Zk Stack chain operator should (might?) be able to set a custom genesis config for their hyperchain.

### Current Behavior

This does not seem possible, or I missed it.

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by zk-Lumi** [comment, marked as answer]
Hi @ETeissonniere, 

Currently custom genesis file for state migration is not supported, but we plan to add support in the future.

**Comment by belanoparo** [comment]
Currently, Zk Sync does not support setting a custom genesis configuration when creating a new hyperchain. The creation of hyperchains in Zk Sync follows a predefined set of rules and configurations, and there is no built-in functionality to customize the genesis configuration. If you require custom configurations for your hyperchain, you may need to explore alternative solutions or reach out to the Zk Sync team for further assistance or guidance on potential future developments in this area.

**Comment by procebetjumorales** [comment]
It is currently not possible to set a custom genesis configuration for a hyperchain in the Zk Stack.

 While this feature may not be supported at the moment, it could potentially be considered for future updates or releases.

 For further clarification or to inquire about future developments, consider reaching out to the Zk Stack development team directly.

Happy coding and exploring with Zk Stack!

**Comment by cerin74** [comment]
Currently, setting a custom genesis configuration for a hyperchain in the ZK Stack is not supported or documented. Further clarification or enhancement of this feature may be required.

**Comment by placy23** [comment]
Customizing genesis configurations for hyperchains in the ZK Stack is currently not supported or documented. Further investigation is required to determine if this feature can be implemented.

**Comment by m4ccmocare** [comment]
The user is unsure if it is possible to set a custom genesis configuration when creating a new hyperchain in the Zk Stack. They expect that a Zk Stack chain operator should be able to do so, but are not sure if it is supported or how to do it. The user has not found any similar issues in their prior research. No repository link was provided.
